### PR TITLE
Fix window decorations

### LINKED ISSUE #10731: Can't escape full screen + menus stay closed (after upgrade to 5.12/5.13)
### JabRef version

5.12 (latest release)

### Operating system

GNU / Linux

### Details on version and operating system

Mint 21.2 with Cinnamon 5.8.4

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Upgrade to 5.12 by manually installing the Debian package.
2. Open JabRef.
3. The window is in full-screen (Cinnamon's title bar and bottom panel aren't visible), and when I click on a menu item (File, Edit, etc.) it appears and disappears immediately after.
4. The solutions from [https://docs.jabref.org/installation](https://docs.jabref.org/installation) (adding Java options) don't change anything.

### Appendix

Notice that I'll be able to perform some tests tomorrow if needed, but then I'll just re-install a previous version to be able to work. :disappointed: 

<details>

<summary>Log File</summary>

```
2023-12-27 18:44:25 [JavaFX-Launcher] sun.util.logging.internal.LoggingProviderImpl$JULWrapper.log()
WARN: Unsupported JavaFX configuration: classes were loaded from 'module org.jabref.merged.module', isAutomatic: false, isOpen: true
2023-12-27 18:44:27 [JavaFX Application Thread] org.jabref.gui.StateManager.setActiveDatabase()
INFO: No open database detected
2023-12-27 18:44:27 [JavaFX Application Thread] org.jabref.gui.theme.ThemeManager.updateThemeSettings()
INFO: Not updating theme because it hasn't changed
2023-12-27 18:44:27 [JavaFX Application Thread] org.jabref.gui.theme.ThemeManager.updateThemeSettings()
INFO: Theme set to Theme{type=EMBEDDED, name='Dark.css'} with base css StyleSheet{jrt:/org.jabref/org/jabref/gui/Base.css}
2023-12-27 18:44:27 [JavaFX Application Thread] org.jabref.gui.theme.ThemeManager.updateThemeSettings()
INFO: Not updating theme because it hasn't changed
2023-12-27 18:44:27 [JavaFX Application Thread] org.jabref.gui.theme.ThemeManager.updateThemeSettings()
INFO: Theme set to Theme{type=EMBEDDED, name='Dark.css'} with base css StyleSheet{jrt:/org.jabref/org/jabref/gui/Base.css}
2023-12-27 18:44:29 [JavaFX Application Thread] org.jabref.preferences.JabRefPreferences.getFetcherKeysFromKeyring()
WARN: JabRef could not open the key store
2023-12-27 18:44:29 [pool-1-thread-1] org.jabref.gui.JabRefDialogService.notify()
INFO: Ouverture en cours: (...)
```

</details>

### PR DESCRIPTION
**Pull request by calixtus**
Refs #10731 

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/JabRefGUI.java [hunk 1]
```diff
@@ -253,12 +252,13 @@ public void onHiding(WindowEvent event) {
...
-        preferences.setPositionY(mainStage.getY());
-        preferences.setSizeX(mainStage.getWidth());
-        preferences.setSizeY(mainStage.getHeight());
+        if (!mainStage.isMaximized()) {
+            preferences.setPositionX(mainStage.getX());
+            preferences.setPositionY(mainStage.getY());
+            preferences.setSizeX(mainStage.getWidth());
+            preferences.setSizeY(mainStage.getHeight());
+        }
         preferences.setWindowMaximised(mainStage.isMaximized());
```
Comment (koppor) [inline src/main/java/org/jabref/gui/JabRefGUI.java:261]: I would drop this line, too.

This should keep the fix at https://github.com/JabRef/jabref/issues/4939

**Review by koppor — CHANGES_REQUESTED** [review]
Reading https://ux.stackexchange.com/a/125593/93436 and https://unix.stackexchange.com/q/43106/18033, I think there should be an **option** whether JabRef should remember its size. Default should be **yes** on Windows and **no** on Linux.

I am even thinking to put "no" as default on Windows, because when users have "FancyZones" installed, the Window management should be better...

**Comment by koppor** [comment]
Note: The `x` values are "somehow" globally for multiple screens:

Example for

image

[javafx.stage.Screen@ede6e680 bounds:Rectangle2D [minX=0.0, minY=0.0, maxX=1600.0, maxY=2560.0, width=1600.0, height=2560.0] visualBounds:Rectangle2D [minX=0.0, minY=156.0, maxX=1600.0, maxY=2560.0, width=1600.0, height=2404.0] dpi:125.0 outputScale:(1.5,1.5), javafx.stage.Screen@42b22ab3 bounds:Rectangle2D [minX=5440.0, minY=96.0, maxX=6880.0, maxY=2656.0, width=1440.0, height=2560.0] visualBounds:Rectangle2D [minX=5440.0, minY=96.0, maxX=6880.0, maxY=2616.0, width=1440.0, height=2520.0] dpi:108.0 outputScale:(1.0,1.0), javafx.stage.Screen@21359912 bounds:Rectangle2D [minX=1600.0, minY=232.0, maxX=5440.0, maxY=2392.0, width=3840.0, height=2160.0] visualBounds:Rectangle2D [minX=1600.0, minY=232.0, maxX=5440.0, maxY=2352.0, width=3840.0, height=2120.0] dpi:139.0 outputScale:(1.0,1.0)]

**Comment by koppor** [comment]
I enabled debug output, added more things to debug output and included also more checks for "out of bounds". Lets see, if this matches more cases.

**Comment by koppor** [comment]
Note that the updated code also should keep https://github.com/JabRef/jabref/issues/5037 fixed.

I did not add some new preference to store the JabRef X and Y position depending on the screen where JabRef "lives". This can be a follow up (if users ask for it)

**Comment by koppor** [comment]
We could always save the screen position if user moves the window. In case the user fixes the window position and then maximizes JabRef and then quits, the user-fixed window position is not stored.

**Comment by Siedlerchr** [comment]
Related https://bugs.openjdk.org/browse/JDK-8326619

**Review by koppor — APPROVED** [review]

**Comment by koppor** [comment]
There was no follow-up on my PR comment. I think, it is good to go as is, since it brings things forward (and unblocks https://github.com/JabRef/jabref/pull/11441).

Therefore, approved and added to the merge queue.

**Pull request merged**